### ISSUE TITLE
An empty sourcesContent array is added to the source map even if --source-map-contents is not used when compiling

### ISSUE THREAD

**Issue by DJDavid98** [sass/node-sass#1045]
Exporting using the following command line:

    node-sass --source-map=. --output-style=compressed --output=css theme.scss theme.min.css

results in a map file that looks like

	{
		"version": 3,
		"file": "theme.min.css",
		"sources": [
			"theme.scss"
		],
		"sourcesContent": [],
		"mappings": "<mappings stripped>",
		"names": []
	}

It's unnecessary to have `sourcesContent` set as an empty array, it should be omitted instead. Furthermore, it breaks the source-map-visualization utitlity, I opened an issue on that project as well (sokra/source-map-visualization#12).

**Comment by saper** [comment]
is this the only problem? not sure this is a real "bug" but you can try to apply my change at https://github.com/sass/libsass/issues/1380 to `libsass` and see if it helps.

**Comment by DJDavid98** [comment]
I'm not quite sure how to apply changes to libsass itself. I just thought I'd bring it up, since it's unnecessary.

**Comment by nvartolomei** [comment]
@saper it is a problem when combining the output with other tools, in my case it is autoprefixer, which dies if this field is present and empty.

```
Fatal error: "_absolutepath_/scss/style.scss" is not in the SourceMap.
```

**Comment by saper** [comment]
Thanks, good to know. You might want to raise this with other tools as well in the spirit of "be conservative what you send, be liberal in what you receive".

**Comment by xzyfer** [comment]
This does feel like an issue with the "other" tools being a lazy, but we can address this as well.

**Comment by talgautb** [comment]
i have some problems with sourcemap in sass, after sass i use postcss (autoprefixer) which show error:
`Fatal error: "path_to/styles/_button.sass" is not in the SourceMap.
`
```
"autoprefixer": "6.1.0",
"grunt": "0.4.5",
"grunt-postcss": "0.7.1",
"grunt-sass": "1.1.0",
```

**Comment by saper** [comment]
@talgautb Can you open another issue? Please read our https://github.com/sass/node-sass/blob/master/TROUBLESHOOTING.md document and provide the information needed. Also please post a script to reproduce the problem without using `grunt-sass` and other plugins; posting a resulting source map can be useful.

**Comment by saper** [comment]
@xzyfer I understand this landed v3.4.0 so I can close this issue?

**Comment by xzyfer** [comment]
This has been fixed in v3.4.0.
